src/lib.rs

```diff
@@ -431,3 +433,89 @@ fn collapse_state_maps(map: &BTreeMap<i64, StateGroupEntry>, state_group: i64) -
 
     state_map
 }
+
+impl Config {
+    pub fn new(
+        db_url: String,
+        output_file: String,
+        room_id: String,
+        max_state_group: String,
+        min_saved_rows: String,
+        transactions: bool,
+        level_sizes: String,
+    ) -> Config {
+        if db_url == "" {
+            panic!("db url is required");
+        }
```

Comment:
Couple of things here:

1. We should return an error rather than panic here, panics should really only be used in cases of "the programmer has done something wrong".
2. I'm not sure its worth trying to do such basic validation here, as:
    a. you have to handle the case where the value is wrong but non-empty anyway
    b. its sometimes not immediately obvious that e.g `db_url` can't be empty.

Reply:
I mainly had this here since when running it from the command line it will fail to run if you don't provide a database URL or a room id (and so calling new() would behave just like parse_arguments())

Therefore if you accidently call it from python without specifying either of those things you get an error at this stage rather than "cannot connect to database":

```
>>> comp.run_compression()
thread '<unnamed>' panicked at 'db url is required', src/lib.rs:451:13
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
pyo3_runtime.PanicException: db url is required
```
I agree that all of the unwrap()'s in that method should probably be excepts though!

Reply:
> I mainly had this here since when running it from the command line it will fail to run if you don't provide a database URL or a room id (and so calling new() would behave just like parse_arguments())

If we have required arguments we should mark them as such, rather than defaulting to the empty string? `comp.run_compression()` should die with a TypeError due to too few arguments

Reply:
Ah yes I remember I had them default to empty strings because I couldn't work out how to only have some of the arguments as optional - will look into it now